### ISSUE TITLE
Error: ReferenceError: window is not defined

### ISSUE THREAD

**Issue by angeldevildev** [stephenscaff/react-animated-cursor#54]
Hi! I was using your react-animated-cursor library everything went well I was able to put the cursor in the website, however it gives me an error that would be:
- error node_modules\react-animated-cursor\dist\index.js (43:30) @ useEventListener
- error ReferenceError: window is not defined
null
I am developing this website in Next.js.
I hope you can help me as soon as possible.
Thank you.

![issues-react-pointer](https://github.com/stephenscaff/react-animated-cursor/assets/122638135/1562a39e-1b14-4cd1-b889-5421a7421302)

**Comment by stephenscaff** [comment]
I think this has to do with need to set this as a front end component, now that next seperates FE and server side components. Let me check out that theroy.

**Comment by stephenscaff** [comment]
Mentioned this in another of your issues, but  i pushed a new version with "use client" defined. Getting "Window is not defined" warnings in server console.

You can prevent this by following the next.js dynamic import method included in docs, the only difference is that you still need to define "use client", so a move could be to create a local component, dynamically import in that with "use client" defined.

For example, create `components/Cursor`

```
'use client'

import dynamic from 'next/dynamic';

const AnimatedCursor = dynamic(() => import('react-animated-cursor'), {
    ssr: false,
});

export default function Cursor() {
  return <AnimatedCursor/>
}
```

Then import that in your app or layout file. That seems to work without any warnings. 

Researching different approaches right now. 

Note that is isolated to Next. Integrated with Vite and Parcel builds without issue. So, def related to Next's SSR and server components as default architecture. 

Give the above approach a go and let me know how it goes.

**Comment by stephenscaff** [comment]
Hey.

Just released `2.10.1`, which includes a rewrite of `useEventListener` hook, as well as some other stuff, to be more next/ssr/client component friendly. 

Now, you shouldn't have to use Next's Dynamic Import anymore. A simple import _should_ work without issue. 

Just spun up a fresh Next build, using app directory, and was able to import in the layout.tsx without errors or warnings. 

Gonna close this for now, but open back up if you have any issues. 

Thanks!
